sdk/servicebus/azure-servicebus/azure/servicebus/_pyamqp/_message_backcompat.py

```diff
@@ -5,16 +5,40 @@
 # --------------------------------------------------------------------------
 
 # pylint: disable=too-many-lines
-from typing import Callable, cast, TYPE_CHECKING
 from enum import Enum
+from typing import (
+    Callable,
+    cast,
+    TYPE_CHECKING,
+    Any,
+    List,
+    Optional,
+    Dict,
+    Union,
+    Protocol,
+    Tuple,
+)
 
 from ._encode import encode_payload
 from .utils import get_message_encoded_size
 from .error import AMQPError
 from .message import Header, Properties
 
 if TYPE_CHECKING:
-    from ..amqp._amqp_message import AmqpAnnotatedMessage
+    from ..amqp._amqp_message import AmqpAnnotatedMessage, AmqpMessageProperties, AmqpMessageHeader
+    from .message import Message
+    from .error import ErrorCondition
+    class Settler(Protocol):
+        def settle_messages(
+            self,
+            delivery_id: Optional[int],
+            outcome: str,
+            *,
+            error: AMQPError = AMQPError,
```

Comment:
If it is default AMQPError wouldn't this be
```suggestion
            error: Type[AMQPError] = AMQPError,
```
which then would throw a type error further down when you do error=AMQPError(.....)?

Reply:
setting it to None as the default here with type Optional[AMQPError]